assets/src/components/form/base/auto-form-row.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External imports
+ */
+import { Component } from 'react';
+import { isArray } from 'lodash';
+
+/**
+ * Internal imports
+ */
+import { mapValidatorsToAttributes } from '../inputs/base/utils';
+
+/**
+ * AutoFormRow
+ * This component determines whether the incoming form element
+ * is an input, label, or row component and responds accordingly.
+ * FormRow components from a form layout strategy will simply be
+ * returned as is, since they require no further manipulation.
+ * Inputs and Labels will be passed to the supplied form layout
+ * strategy AutoColumnRow component for further processing.
+ * If a Label component is not provided but the supplied Input
+ * has a "label" prop, then that will be used to generate an
+ * InputLabel using the supplied form layout strategy component.
+ *
+ * @component
+ * @param {Object} FormElement either an input, label, or form row component
+ * @param {Object} InputLabel form layout strategy label display component
+ * @param {Object} AutoColumnRow form layout strategy row component
+ * @return {Object} rendered form row
+ */
+export class AutoFormRow extends Component {
+	render() {
+		const {
+			FormElement,
+			InputLabel,
+			AutoColumnRow,
+		} = this.props;
+
+		let label = null;
+		let input = null;
+		let row = null;
+		if ( FormElement.type ) {
+			const name = FormElement.type.name;
+			if ( name === 'FormRow' ) {
+				return FormElement;
+			}
+			if ( name === 'InputLabel' ) {
+				label = FormElement;
+			}
+			if ( name === 'FormInput' || name === 'Field' ) {
+				input = FormElement;
+				const validations = isArray( FormElement.props.validations ) ?
+					FormElement.propsvalidations :
+					[ FormElement.props.validations ];
+				let attributes = Object.assign( {}, FormElement.props );
+				attributes = mapValidatorsToAttributes(
+					attributes,
+					validations
+				);
```

Comment:
But won't that spread the props out into individual variables?

I guess I could change `mapValidatorsToAttributes()` to handle that but (and I could  easily be wrong) your suggestion doesn't appear to be functionally equivalent as the code it would be replacing (but again, would work perfectly well with a corresponding change to `mapValidatorsToAttributes()`).

I'll give this a try...

Reply:
it will spread the props out into a new object.  It's functionally equivalent to what you did, just more concise.